mam/api/tests/test_api.c

```diff
@@ -445,6 +450,28 @@ static void test_api_save_load() {
   mam_api_destroy(&loaded_api);
 }
 
+static void test_api_save_load_wrong_key() {
+  mam_api_t loaded_api;
+
+  tryte_t encryption_key_trytes[81] = {
+      "NOPQRSTUVWXYZ9ABCDEFGHIJKLM"
+      "NOPQRSTUVWXYZ9ABCDEFGHIJKLM"
+      "NOPQRSTUVWXYZ9ABCDEFGHIJKLM"};
+
+  // Very slightly different key
+  tryte_t decryption_key_trytes[81] = {
+      "MOPQRSTUVWXYZ9ABCDEFGHIJKLM"
+      "NOPQRSTUVWXYZ9ABCDEFGHIJKLM"
+      "NOPQRSTUVWXYZ9ABCDEFGHIJKLM"};
+
+  TEST_ASSERT(mam_api_save(&api, "mam-api.bin", encryption_key_trytes, 81) == RC_OK);
+  // Theoretically, wrong key can lead to decrypting a ciphertext to wrong values
+  // which are not necessarely invalid, but chance is slim (RC_MAM_INVALID_VALUE)
+  TEST_ASSERT(mam_api_load("mam-api.bin", &loaded_api, decryption_key_trytes, 81) != RC_OK);
```

Comment:
What is the error code here ? `RC_MAM_INVALID_VALUE` ?

Reply:
yes